File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/DTO/DTOMagazine.java
package abbruzzese.webpage.webpage.server.webpage.DTO;

import java.time.LocalDate;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DTOMagazine {

    private String Nombre;

    private String Descripcion;

    private String Año;

    private String imageString;

    private String enlace;

    private LocalDate fecha;

    public DTOMagazine(){}

    public DTOMagazine(String Nombre,String Descripcion,String Año,String imageString,String link,LocalDate fecha){

        this.Nombre=Nombre;
        this.Descripcion=Descripcion;
        this.Año=Año;
        this.imageString=imageString;
        this.enlace=link;
        this.fecha=fecha;
    }

    public DTOMagazine(String Nombre,String enlace,String imageString){
        this.Nombre=Nombre;
        this.enlace=enlace;
        this.imageString=imageString;
    }

    public String getNombre() {
        return Nombre;
    }

    public void setNombre(String nombre) {
        Nombre = nombre;
    }

    public String getDescripcion() {
        return Descripcion;
    }

    public void setDescripcion(String descripcion) {
        Descripcion = descripcion;
    }

    public String getAño() {
        return Año;
    }

    public void setAño(String año) {
        Año = año;
    }

    public String getImageString() {
        return imageString;
    }

    public void setImageString(String imageString) {
        this.imageString = imageString;
    }

    public String getEnlace() {
        return enlace;
    }

    public void setEnlace(String enlace) {
        this.enlace = enlace;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

}

File: PHP_API/DTO/DTOMagazine.php
<?php 

namespace App\DTO;

use JsonSerializable;

class DTOMagazine implements JsonSerializable{

    public $nombre;

    public $descripcion;

    public $año;

    public $imagestring;

    public $enlace;

    public $fecha;

    public function __construct($nombre,$descripcion,$año,$imagestring,$enlace,$fecha){
    
        $this->nombre=$nombre;
        $this->descripcion=$descripcion;
        $this->año=$año;
        $this->imagestring=$imagestring;
        $this->enlace=$enlace;
        $this->fecha=$fecha;

    }    

    //Getters y Setters

    public function getNombre(){return $this->nombre;}
    public function setNombre($nombre){return $this->nombre=$nombre;}

    public function getDescripcion(){return $this->descripcion;}
    public function setDescripcion($descripcion){return $this->descripcion=$descripcion;}

    public function getAño(){return $this->año;}
    public function setAño($año){$this->año=$año;}

    public function getimagestring(){return $this->imagestring;}
    public function setimagestring($imagestring){return $this->imagestring=$imagestring;}

    public function getEnlace(){return $this->enlace;}
    public function setEnlace($enlace){return $this->enlace=$enlace;}

    public function getFecha(){return $this->fecha;}
    public function setFecha($fecha){$this->fecha=$fecha;}

    public function jsonSerialize(): array
    {
        return [
            'nombre' => $this->nombre,
            'descripcion' => $this->descripcion,
            'año' => $this->año,
            'imagestring' => $this->imagestring,
            'enlace' => $this->enlace,
            'fecha' => $this->fecha,
        ];
    }
}

?>